### PR TITLE
fix enhance envoy readiness probe

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixed Envoy readiness probe which was reverted in https://github.com/istio/istio/pull/17505

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn Found couple of minor things and fixed them
- When checking for port earlier was using Dial without timeout - introduced timeout now
- For all listeners it was trying to ping on LocalAddress but now changed to it to use whatever listener address is.

Can you PTAL? Also Is there a way to test this if it actually resolved the issue?

**Comment by ramaraochavali** [comment]
As I have mentioned above, made couple of changes which may help - was suspecting Dial with out timeouts may be causing some
Issue. I am not 100% certain about whether these will fix. That is I was asking if there is a way to test this without merging?
A

**Comment by howardjohn** [comment]
sorry didn't see your comment.. last time on your pr immediately after merging a bunch of tests started failing 20% of time or so, so the fact that they didn't and no retest is a good sign. I ran across this by having an empty cluster and spinning up a couple pods, immediately those used high cpu as well as mixer. So I can test that out again with this pr

**Comment by ramaraochavali** [comment]
@howardjohn Great. Thanks. Will wait for you test then.

File: pilot/cmd/pilot-agent/status/util/listeners.go [hunk 1]
```diff
@@ -106,3 +107,24 @@ func getLocalIPPrefixes() []string {
...
+	}
+
+	listeners := &admin.Listeners{}
+	if err := jsonpb.Unmarshal(buf, listeners); err != nil {
+		return nil, fmt.Errorf("failed parsing Envoy listeners %s: %s", buf, err)
+	}
+	addrs := make([]string, 0)
+	for _, l := range listeners.ListenerStatuses {
+		if l.Name == networking.VirtualOutboundListenerName || l.Name == networking.VirtualInboundListenerName {
+			addrs = append(addrs, fmt.Sprintf("%s:%d", l.LocalAddress.GetSocketAddress().GetAddress(), l.LocalAddress.GetSocketAddress().GetPortValue()))
```
Comment (nmittler) [inline pilot/cmd/pilot-agent/status/util/listeners.go]: You should probably be a bit more careful here. Check for `nil` along the way.

**Comment by ramaraochavali** [comment]
@howardjohn Thanks. I will try to reproduce this locally. for now, I am closing this.

**Pull request closed without merging**